kicksaw_integration_utils/aws/sqs/queue.py

```diff
@@ -0,0 +1,339 @@
+
+        Returns
+        -------
+        handles : List[str]
+            List of message handles. Used later to remove the messages from the queue.
+        messages : List[pydantic.BaseModel]
+            List of messages serialized to the provided pydantic model.
+
+        """
+        poll_attempts: int = 0
+        handles: List[str] = []
+        messages: List[PydanticModel] = []
+        while len(messages) < max_messages:
+            # TODO- handle SQS.Client.exceptions.OverLimit
+            response = self._queue.receive_messages(
+                MaxNumberOfMessages=10,
+                WaitTimeSeconds=wait_time_seconds,
+            )
+            logger.debug("Received %d messages from the queue", len(response))
+            for message in response:
+                handles.append(message.receipt_handle)
+                messages.append(self._message_model.parse_raw(message.body))
+
+            if len(response) < 10:
+                poll_attempts += 1
+                if poll_attempts >= max_poll_attempts:
+                    logger.debug("%s has no messages left", self.name)
+                    break
+
+        return handles, messages
+
+    def delete_messages(self, handles: List[str]) -> List[bool]:
+        """
+        Delete messages from the queue using their handles.
+
+        Parameters
+        ----------
+        handles : List[str]
+            Message handles.
+            See .receive_messages method.
+
+        Returns
+        -------
+        List[bool]
+            List of deletion results (True for success, False for failure).
+
+        Raises
+        ------
+        AssertionError
+            Number of messages reported to be deleted doesn't match number of messages
+            requested to be deleted.
+
+        """
+        logger.debug(
+            "Deleting %d messages from %s in batches of 10",
+            len(handles),
+            self.name,
+        )
+
+        results: List[bool] = []
+        for i in range(0, len(handles), 10):
+            batch = handles[i : i + 10]
+            logger.debug(
+                "Deleting batch of %d messages from %s",
+                len(batch),
+                self.name,
+            )
+            response = self._queue.delete_messages(
+                Entries=[
+                    {"Id": f"{i+j}", "ReceiptHandle": handle}
+                    for j, handle in enumerate(batch)
+                ]
+            )
+
+            # Parse response
+            batch_results = sorted(
+                [
+                    *[
+                        {"id": int(entry["Id"]), "success": True}
+                        for entry in response.get("Successful", [])
+                    ],
+                    *[
+                        {"id": int(entry["Id"]), "success": False}
+                        for entry in response.get("Failed", [])
+                    ],
+                ],
+                key=lambda x: x["id"],
+            )
```

Comment:
could this be abstracted into a `staticmethod`? It's the same code as above, right?

Reply:
I thought about this, but decided not to because it's a single expression (code surrounding it is different) and moving it to a separate helper function would actually make the methods less readable. I suggest making it a standalone function if we see this pattern elsewhere.